### PR TITLE
Feature/pelagos 4393 add rename file message handler

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/MessageHandler/RenameFileHandler.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        $this->logger->info(sprintf('Rename File worker started with ID: %d', $fileId));
+        $file = $this->fileRepository->find($fileId);
+        if ($file instanceof File) {
+            $filePhysicalPath = $file->getPhysicalFilePath();
+            if ($file->getStatus() === File::FILE_DELETED) {
+                $this->logger->info(sprintf('Marking File as deleted for ID: %d', $fileId));
+                $newFilePath = $filePhysicalPath . Datastore::MARK_FILE_AS_DELETED;
+                $newFilePath = $this->dataStore->renameFile($filePhysicalPath, $newFilePath, true);
+            } else {
+                $this->logger->info(sprintf('File is renamed on disk for ID: %d', $fileId));
```
Comment (fightingtexasaggie) [inline src/MessageHandler/RenameFileHandler.php:83]: perhaps add the word "already" before renamed.

File: src/Controller/Api/FileManager.php [hunk 2]
```diff
@@ -106,8 +108,12 @@ public function addFile(DatasetSubmission $datasetSubmission, Request $request,
      *
      * @return Response
      */
-    public function delete(DatasetSubmission $datasetSubmission, Request $request, EntityManagerInterface $entityManager, Datastore $datastore): Response
-    {
+    public function delete(
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:111]: Shouldn't this bracket be on the next line?

File: src/Controller/Api/FileManager.php [hunk 3]
```diff
@@ -155,21 +161,19 @@ private function deleteFile(File $file, Fileset $fileset, Datastore $datastore)
             }
         } elseif ($file->getStatus() === File::FILE_DONE) {
             $file->setStatus(File::FILE_DELETED);
-            $filePath = $file->getPhysicalFilePath();
-            $newFilePath = $filePath . Datastore::MARK_FILE_AS_DELETED;
-            $newFilePath = $datastore->renameFile($filePath, $newFilePath, true);
-            $file->setPhysicalFilePath($newFilePath);
+            $markAsDeleted = new RenameFile($file->getId());
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:164]: Maybe the variable name here should be renameMessage?

File: src/Controller/Api/FileManager.php [hunk 4]
```diff
@@ -227,22 +231,22 @@ public function updateFileOrFolderName(
...
+     * @param Fileset             $fileset     Fileset entity instance.
+     * @param string              $newFileName New file name for the file.
+     * @param MessageBusInterface $messageBus  Message bus interface.
      *
      * @throws BadRequestHttpException When the destination file name already exists.
      *
      * @return void
      */
-    private function updateFileName(File $file, Fileset $fileset, string $newFileName, Datastore $datastore) : void
+    private function updateFileName(File $file, Fileset $fileset, string $newFileName, MessageBusInterface $messageBus) : void
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:243]: Fileset doesn't really have to be passed anymore, you can get it from File.

**Review by mickel1138 — APPROVED** [review]
Tested, and works. Minor changes suggested.

File: src/Controller/Api/FileManager.php (hunk 3 above)
Comment (praneethpr) [reply src/Controller/Api/FileManager.php:164]: Sure

**Review by fightingtexasaggie — COMMENTED** [review]
renaming a file in dataset review resulted in the file being copied out of the UDI directory, without the name change applied.

File: src/Controller/Api/FileManager.php [hunk 5]
```diff
@@ -137,13 +143,13 @@ public function delete(DatasetSubmission $datasetSubmission, Request $request, E
...
-     * @param Fileset   $fileset   Fileset entity instance.
-     * @param Datastore $datastore Datastore to manipulate the file on disk.
+     * @param File                $file       File entity that needs to be deleted.
+     * @param Fileset             $fileset    Fileset entity instance.
+     * @param MessageBusInterface $messageBus Message bus interface.
      *
      * @return void
      */
-    private function deleteFile(File $file, Fileset $fileset, Datastore $datastore) : void
+    private function deleteFile(File $file, Fileset $fileset, MessageBusInterface $messageBus) : void
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:152]: Fileset doesn't really have to be passed... But it can stay.

**Review by mickel1138 — APPROVED** [review]
Tested, and renames correctly and also marks deleted files correctly.

**Review by fightingtexasaggie — COMMENTED** [review]
When making filename changes in dataset review, the zipfile isn't being updated with the latest filename changes.

**Review by fightingtexasaggie — APPROVED** [review]
confirmed zipfile staleness fixed in other branch, so is good to go

**Pull request merged**